### ISSUE TITLE
Uploads and downloads page could be very large

### ISSUE THREAD

**Issue by vitoyucepi** [slskd/slskd#1143]
## Summary

By default, data retention is not set for downloads and uploads.
https://github.com/slskd/slskd/blob/960f03d59523c84a7f1fc256cbfacf541513d18a/docs/config.md?plain=1#L889
This can lead to huge memory consumption on downloads and uploads pages because all data is displayed on a single page. Also, the entire list is transferred every second.
https://github.com/slskd/slskd/blob/960f03d59523c84a7f1fc256cbfacf541513d18a/src/web/src/components/Transfers/Transfers.jsx#L27-L41
It may be hurtful to view these pages if you are using a metered or limited connection.

## Actual behavior

When browsing uploads and downloads pages without data retention, they can use a lot of memory and transfer a lot of data.
They are also not very informative.

## Expected behavior

1. Don't update the downloads/uploads log every second.
1. Add categories and pagination.

## Versions

- slskd: *0.21.0.0*.

**Comment by jpdillingham** [comment]
I agree this page (and others, like the browse interface) need to be redesigned due to poor performance.  I haven't yet had an opportunity to focus on this.

In the meantime there are buttons on the System screen that you can use to clear these lists if you get into a state where the pages won't load:

![image](https://github.com/slskd/slskd/assets/17145758/41d60cd9-6666-4ae7-8baf-5f15c96abbdc)

I also added [retention settings](https://github.com/slskd/slskd/blob/master/docs/config.md#data-retention) that will cause the app to automatically clear old records.  This won't help much if active transfers are what's causing you issues.

If anyone has any suggestions regarding how a newly designed screen would look and behave I would appreciate it.  Better if you have mockups!

**Comment by vitoyucepi** [comment]
The first thing I will probably do is enable the retention period and reduce the update frequency.
Second, I'll implement two pages for download and upload queues and disable updates for download and upload history.

**Comment by neofright** [comment]
Putting a +1 here as I have no intention of ever clearing my transfers database. With this many transfers the uploads page never loads (it probably would eventually but I've never waited long enough):

```
sqlite> SELECT COUNT(*) FROM Transfers WHERE StateDescription = 'Completed, Succeeded' AND Direction = 'Upload';
352504
```
Pagination or an SQL queries that default to perhaps the last 7 days, then the last months etc. would be a nice solution to this. For now, I just do:
```
"SELECT RequestedAt,Username,Filename FROM Transfers WHERE StateDescription = 'Completed, Succeeded' AND Direction = 'Upload' AND RequestedAt LIKE strftime('%Y', 'now') || '%'
```
when I want to check all of the transfers for this year.

**Comment by Uno-Muno** [comment]
Dabbled a bit in design. Feel free to critique/dismiss.

First, a better overview should be given:
<img width="960" height="807" alt="Image" src="https://github.com/user-attachments/assets/86334f19-ec08-49b9-9b9f-6084129ad076" />

For this, all the users should be collapsed with only overview level info visible (if something has failed or was cancelled, some info about folders). Active downloads/uploads are of highest importance in these views therefore require some special treatment - a filter, a section, tabs. Unless user has many concurrent downloads section (as seen on the mockup above) is most simple.

If user wants more info or granular control, they can open up the accordions:
<img width="960" alt="Open folders" src="https://github.com/user-attachments/assets/d364146a-088b-4899-b382-2aee773bc832" />

Warning, opinion: I removed most of the stroke - they make things more cluttered if overused and I feel it's currently the case. For better line tracking some slight hover effects can be used instead:
<img width="960" alt="Hover effect" src="https://github.com/user-attachments/assets/0ab9b50d-e7f2-4308-9a14-8df63a30202a" />

Not sure how how to handle actions. Duplication of buttons for delete, retry etc. may make things more complex, but it also gives more options.

**Comment by jpdillingham** [comment]
> Putting a +1 here as I have no intention of ever clearing my transfers database. With this many transfers the uploads page never loads (it probably would eventually but I've never waited long enough):

Configuring retention or using the button on the data tab only marks the records as "removed", so it's a soft delete.  There's no logic to remove transfer records currently.

**Comment by neofright** [comment]
> Configuring retention or using the button on the data tab only marks the records as "removed", so it's a soft delete. There's no logic to remove transfer records currently.

I should clarify that I meant doing so in an unsupported way such as `DELETE FROM Transfers` / `TRUNCATE TABLE Transfers`.
